crates/matrix-sdk/tests/integration/matrix_auth.rs

```diff
@@ -313,3 +320,157 @@ fn test_serialize_session() {
         })
     );
 }
+
+#[async_test]
+async fn test_login_with_cross_signing_bootstrapping() {
+    let (builder, server) = test_client_builder().await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/r0/login"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::LOGIN))
+        .mount(&server)
+        .await;
+
+    Mock::given(method("POST"))
+        .and(path("/_matrix/client/r0/keys/query"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(json!({
+            "device_keys": {
+                "@cheeky_monkey:matrix.org": {}
+            }
+        })))
+        .mount(&server)
+        .await;
+
+    // TODO why unstable?
```

Comment:
Because we didn't fetch the supported versions from the server, so it falls back to the first supported version of this endpoint.

Once we have the versions, Ruma will use a newer variant of the endpoint. If you'd like to see this here, you'll have to set the versions manually or mock the [`/versions`] endpoint.

[`/versions`]: https://spec.matrix.org/unstable/client-server-api/#get_matrixclientversions